test/serviceContext-test.js

```diff
@@ -0,0 +1,252 @@
+  };
+
+  let res6 = ci.verifyAlgSupport('RS256', 'id_token', 'signing_alg');
+  it('verify_alg_support', function() {
+    assert.isTrue(res6);
+  });
+
+  let res7 = ci.verifyAlgSupport('RS512', 'id_token', 'signing_alg');
+  it('verify_alg_support', function() {
+    assert.isFalse(res7);
+  });
+
+  let res8 = ci.verifyAlgSupport('RSA1_5', 'userinfo', 'encryption_alg');
+  it('verify_alg_support', function() {
+    assert.isTrue(res8);
+  });
+
+  let res9 = ci.verifyAlgSupport('ES256', 'token_endpoint_auth', 'signing_alg');
+  it('verify_alg_support', function() {
+    assert.isTrue(res9);
+  });
+
+  ci.providerInfo['issuer'] = 'https://example.com/';
+  let url_list = ci.generateRequestUris('/leading');
+  let sp = urlParse(url_list[0]);
+  let p = sp.pathname.split('/');
+
+  it('verify_requests_uri', function() {
+    assert.deepEqual(p[0], '');
+    assert.deepEqual(p[1], 'leading');
+    assert.deepEqual(p.length, 3);
+  });
+
+  ci.providerInfo['issuer'] = 'https://op.example.org/';
+  url_list = ci.generateRequestUris('/leading');
+  sp = urlParse(url_list[0]);
+  let np = sp.pathname.split('/');
+
+  it('verify_requests_uri test2', function() {
+    assert.deepEqual(np[0], '');
+    assert.deepEqual(np[1], 'leading');
+    assert.deepEqual(np.length, 3);
+    assert.notDeepEqual(np[2], p[2]);
+  });
+});
+
+describe('client info tests', function() {
+  let config;
+  let ci;
+  beforeEach(function() {
+    config = {
+      'client_id': 'client_id',
+      'issuer': 'issuer',
+      'client_secret': 'client_secret',
+      'base_url': 'https://example.com',
+      'requests_dir': 'requests'
+    };
+
+    ci = new ServiceContext(null, config);
+  });
+
+  it('client info init', function() {
+    assert.isNotNull(ci);
+    if (Object.keys(config).indexOf('client_id') > -1) {
+      assert.deepEqual(ci.config['client_id'], config['client_id']);
+    } else if (Object.keys(config).indexOf('issuer') > -1) {
+      assert.deepEqual(ci.config['issuer'], config['issuer']);
+    } else if (Object.keys(config).indexOf('client_secret') > -1) {
+      assert.deepEqual(ci.config['client_secret'], config['client_secret']);
+    } else if (Object.keys(config).indexOf('base_url') > -1) {
+      assert.deepEqual(ci.config['base_url'], config['base_url']);
+    } else if (Object.keys(config).indexOf('request_dir') > -1) {
+      assert.deepEqual(ci.config['request_dir'], config['request_dir']);
+    }
+  });
+});
+
+describe('set and get client secret', function() {
+  let ci;
+  beforeEach(function() {
+    ci = new ServiceContext();
+    ci.clientSecret = 'supersecret';
+  });
+
+  it('client info init', function() {
+    assert.deepEqual(ci.clientSecret, 'supersecret');
+  });
+});
+
+describe('set and get client id', function() {
+  let ci;
+  beforeEach(function() {
+    ci = new ServiceContext();
+    ci.clientId = 'myself';
+  });
+
+  it('client info init clientId', function() {
+    assert.deepEqual(ci.clientId, 'myself');
```

Comment:
What is the intention of this test?
It should literally always pass, unless you decide using the new ES6 set/get class syntax. Then it might fail depending on your implementation.

Also it is not `clientId`, but `client_id`.